modules/ppcp-wc-gateway/resources/js/pay-upon-invoice.js

```diff
@@ -1,34 +1,43 @@
 window.addEventListener('load', function() {
 
-    const getSessionIdFromJson = () => {
-        const form = document.querySelector('form.checkout');
-        if(!form) {
-            return;
-        }
+    function _loadBeaconJS(options) {
+        var script = document.createElement('script');
+        script.src = options.fnUrl;
+        document.body.appendChild(script);
+    }
 
-        const fncls = document.querySelector("[fncls='fnparams-dede7cc5-15fd-4c75-a9f4-36c430ee3a99']");
-        if(!fncls) {
-            return;
+    function _injectConfig() {
+        var script = document.querySelector("[fncls='fnparams-dede7cc5-15fd-4c75-a9f4-36c430ee3a99']");
+        if (script) {
+            if (script.parentNode) {
+                script.parentNode.removeChild(script);
+            }
         }
-        const fncls_params = JSON.parse(fncls.textContent);
 
-        if(document.querySelector("[name='fraudnet-session-id']") !== null) {
-            document.querySelector("[name='fraudnet-session-id']").remove();
-        }
+        script = document.createElement('script');
+        script.id = 'fconfig';
+        script.type = 'application/json';
+        script.setAttribute('fncls', 'fnparams-dede7cc5-15fd-4c75-a9f4-36c430ee3a99');
 
-        const fraudnetSessionId = document.createElement('input');
-        fraudnetSessionId.setAttribute('type', 'hidden');
-        fraudnetSessionId.setAttribute('name', 'fraudnet-session-id');
-        fraudnetSessionId.setAttribute('value', fncls_params.f);
+        var configuration = {
+            'f': FraudNetConfig.f,
+            's': FraudNetConfig.s
+        };
 
-        form.appendChild(fraudnetSessionId);
-        console.log(fncls_params)
+        script.text = JSON.stringify(configuration);
+        document.body.appendChild(script);
+
+        _loadBeaconJS({fnUrl: "https://c.paypal.com/da/r/fb.js"})
     }
 
     document.addEventListener('hosted_fields_loaded', (event) => {
-        getSessionIdFromJson();
+        if (PAYPAL.asyncData && typeof PAYPAL.asyncData.initAndCollect === 'function') {
+            PAYPAL.asyncData.initAndCollect()
+        }
+
+        _injectConfig();
     });
 
-    getSessionIdFromJson();
+    _injectConfig();
```

Comment:
Is it always executed, even if the store does not support pay upon invoice?

Reply:
Theoretically nothing should neither load nor executed when store is not `DE` but maybe I´m missing something here?
https://github.com/woocommerce/woocommerce-paypal-payments/blob/PCP-417-new-feature---pay-upon-invoice/modules/ppcp-wc-gateway/src/WCGatewayModule.php#L192-L193